megatron/fused_kernels/layer_norm_cuda_kernel.cu

```diff
@@ -304,7 +304,11 @@ void cuApplyLayerNorm(
   // 1) blockDim.x == warpSize
   // 2) Tensors are contiguous
   //
+#ifndef __HIP_PLATFORM_HCC__
   for (auto i1=blockIdx.y; i1 < n1; i1 += gridDim.y) {
+#else
+  for (int i1=blockIdx.y; i1 < n1; i1 += gridDim.y) {
```

Comment:
@rraminen Can you please mention the error message you see without this change on ROCm?

Reply:
/root/Megatron-DeepSpeed/megatron/fused_kernels/layer_norm_hip_kernel.hip:547:40: error: no viable overloaded '+='
for (auto i1=blockIdx.y; i1 < n1; i1 += gridDim.y) {
~~ ^ ~~~~~~~~~
/opt/rocm-5.1.3/hip/include/hip/amd_detail/amd_hip_complex.h:234:1: note: candidate function not viable: no known conversion from 'typename __HIP_Coordinates<__HIP_BlockIdx>::__Y' to 'hipFloatComplex &' (aka 'HIP_vector_type<float, 2> &') for 1st argument
COMPLEX_ADD_PREOP_OVERLOAD(hipFloatComplex)

Reply:
@jeffdaily This error doesn't make sense to me. It looks like the statement in question 
`(auto i1=blockIdx.y; i1 < n1; i1 += gridDim.y) {`
should be compilable using `auto` keyword. Should this be a JIRA for the HIP team?

Reply:
It doesn’t make sense to me either, but I’ve seen this a few other times and I have solved it exactly the same way by using int instead of auto.  Please raise a ticket so we can get some clarity from HIP team.

Reply:
Filed JIRA: https://ontrack-internal.amd.com/browse/SWDEV-348565